### PR TITLE
feat/CIV-1774: Claimant Issued Email for 1v2 Two Legal Reps

### PR DESCRIPTION
**Pull request by HarryH96**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-1774

### Change description ###
- Adding logic to handler for sending multiparty template + fields
- New template added for Claimant Solicitor: 1v2 for Claim Issued
ID: `1d734510-6a06-4372-aff2-d5da8bf19f72`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimContinuingOnlineApplicantForSpecNotificationHandler.java [hunk 1]
```diff
@@ -67,9 +69,13 @@ private CallbackResponse notifyApplicantSolicitorForClaimContinuingOnline(Callba
 
         CaseData caseData = callbackParams.getCaseData();
 
+        String emailTemplateID = getMultiPartyScenario(caseData).equals(ONE_V_TWO_TWO_LEGAL_REP)
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimContinuingOnlineApplicantForSpecNotificationHandler.java:72]: Is this ticket only about 1 v 2 2 legal reps? 
If we have 1 v 2 1 legal rep then also I think the email should contain both defendants name.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimContinuingOnlineApplicantForSpecNotificationHandler.java [hunk 2]
```diff
@@ -79,14 +85,33 @@ private CallbackResponse notifyApplicantSolicitorForClaimContinuingOnline(Callba
...
-            CLAIM_LEGAL_ORG_NAME_SPEC, getApplicantLegalOrganizationName(caseData.getApplicant1OrganisationPolicy()
-                .getOrganisation().getOrganisationID(), caseData),
-            CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference(),
-            ISSUED_ON, formatLocalDate(caseData.getIssueDate(), DATE),
-            RESPONDENT_NAME, getPartyNameBasedOnType(caseData.getRespondent1()),
-            RESPONSE_DEADLINE, formatLocalDateTime(caseData.getRespondent1ResponseDeadline(), DATE_TIME_AT)
-        );
+
+        if (getMultiPartyScenario(caseData).equals(ONE_V_TWO_TWO_LEGAL_REP)) {
+            return Map.of(
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimContinuingOnlineApplicantForSpecNotificationHandler.java:90]: In the return statement only difference in the Map is the respondent name, other things are same.
Could we create the map before add the common fields and then add names based on the condition.
Then we can return the final map.

Comment (HarryH96) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimContinuingOnlineApplicantForSpecNotificationHandler.java:90]: Unfortunately the multiparty gov notify template contains a different name for the Defendant One Name:
if its 1v1: RESPONDENT_NAME / "defendantName"
if its 1v2: RESPONDENT_ONE_NAME / "defendantOneName"and RESPONDENT_TWO_NAME / "defendantTwoName"

**Review by jaberelfasi — APPROVED** [review]

**Review by sampankumar — DISMISSED** [review]

**Comment by HarryH96** [comment]
Changes for this PR are included in branch: https://github.com/hmcts/civil-service/pull/945/

**Pull request closed without merging**